File: 1364-tuple-with-same-product/1364-tuple-with-same-product.java
class Solution {
    public int tupleSameProduct(int[] nums) {
        Map<Integer,Integer>mpp = new HashMap<>();
        int count = 0 ;
        for(int i = 0 ; i < nums.length ; i++)
        {
            for(int j = i+1 ; j < nums.length ; j++)
            {
               int product = nums[i] * nums[j] ;
               count += mpp.getOrDefault(product,0);
               mpp.put(product,mpp.getOrDefault(product,0)+1);
            }
        }
        return count* 8 ; 
    }
}

File: 1364-tuple-with-same-product/1364-tuple-with-same-product.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        unordered_map<int, int> productCount;
        int count = 0;
        
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int product = nums[i] * nums[j];
                count += productCount[product]; 
                productCount[product]++;  // Increment product count
            }
        }
        
        return count * 8;
    }
};